clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -171,6 +195,7 @@ public void removeSensor(String name) {
                     }
                 }
             }
+            log.debug("Removed sensor with name {}", name);
```

Comment:
Similar to the other logging statement, I think it's a bit misleading as it will be logged even if this invocation loses the race condition.